### PR TITLE
Added GameActivityTest #85

### PR DESCRIPTION
**Pull request by opticod**
Here is one more UI based test for GameActivity #85

### CODE REVIEW

**Comment by opticod** [comment]
@alma-castant This is a Espresso based UI testing class of `GameActivity` .This will test different Views,adapter and Buttons shown up in different orientations including landscape and portrait ,and the working of different click events on Buttons and in adapter view items.

File: PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * Created by opticod(anupam) on 6/2/16.
+ */
```
Comment (alma-castant) [inline PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java]: Please remove this as we don't use it

File: PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class GameActivityTest {
+    @Rule
+    public ActivityTestRule<GameActivity> activityRule = new ActivityTestRule<>(
+            GameActivity.class);
+
+
```
Comment (alma-castant) [inline PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java]: Leave one empty line only

File: PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    public void viewTest() {
+
+        onView(withId(R.id.scenarioNameEditText)).check(matches(isDisplayed()));
+        activityRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
```
Comment (alma-castant) [inline PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java]: Why are you changing the orientation after doing the check and not before?

File: PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        onView(withId(R.id.clothImageView)).check(matches(isDisplayed()));
+        activityRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
+        onView(withId(R.id.clothImageView)).check(matches(isDisplayed()));
+        activityRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+
+    }
+
+    @Test
+    public void clickEvents() {
+
```
Comment (alma-castant) [inline PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java]: Remove this empty line

File: PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        onData(is(instanceOf(java.lang.String.class)))
+                .inAdapterView(withId(R.id.mainListView))
+                .atPosition(0)
+                .check(matches(isDisplayed())).perform(click());
+
+        onView(withId(R.id.redoButton)).perform(click());
+        onView(withId(R.id.continueButtonGoesToMap)).perform(click());
+
```
Comment (alma-castant) [inline PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java]: Remove this empty line

File: PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    @Test
+    public void clickEvents() {
+
+        onData(is(instanceOf(java.lang.String.class)))
+                .inAdapterView(withId(R.id.mainListView))
+                .atPosition(0)
+                .check(matches(isDisplayed())).perform(click());
+
+        onView(withId(R.id.redoButton)).perform(click());
+        onView(withId(R.id.continueButtonGoesToMap)).perform(click());
```
Comment (alma-castant) [inline PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java]: After perfoming these clicks, you should check their effect

File: PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        onView(withId(R.id.hairImageView)).check(matches(isDisplayed()));
+        activityRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
+        onView(withId(R.id.hairImageView)).check(matches(isDisplayed()));
+        activityRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+
+        onView(withId(R.id.clothImageView)).check(matches(isDisplayed()));
+        activityRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
+        onView(withId(R.id.clothImageView)).check(matches(isDisplayed()));
+        activityRule.getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+
```
Comment (alma-castant) [inline PowerUp/app/src/androidTest/java/powerup/systers/com/powerup/espresso/GameActivityTest.java]: Remove this empty line

**Comment by aanchal07** [comment]
Changes pending since very long so closing.

**Pull request closed without merging**